### PR TITLE
Cherry-pick Add and enforce contracts.allowSystemUseOfHapiSigs

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
- Adds contracts.allowSystemUseOfHapiSigs property, a comma-separated list of zero or more HAPI operations represented by precompiles.
- Unless a precompile's operation appears in the list, it cannot activate signing requirements using the top-level `SignatureMap`.
- Validates end-to-end behavior in HapiSpecs.

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java [hunk 1]
```diff
@@ -223,6 +223,7 @@ public Optional<Throwable> execFor(final HapiSpec spec) {
                 updateStateOf(spec);
             }
         } catch (final Throwable t) {
+            t.printStackTrace();
```
Comment (tinker-michaelj) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java:226]: This was unintentional, though I don't know if we still have any concerns about log spam.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java]: removed.

**Review by mhess-swl — APPROVED** [review]

**Comment by tinker-michaelj** [comment]
Looks like some property updates are "leaking" between tests in CI; we'll have to go through one-by-one to find where. 🧐

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @lukelee-sl. We'll just need to hunt down the property "leak" in CI before going to `develop`.

**Pull request merged**